src/BuiltInTools/dotnet-watch/HotReloadDotNetWatcher.cs

```diff
@@ -129,7 +129,12 @@ public async Task WatchAsync(DotNetWatchContext context, CancellationToken cance
 
                         if (finishedTask != fileSetTask || fileSetTask.Result is not FileItem[] fileItems)
                         {
-                            // The app exited.
+                            if (processTask.IsFaulted && finishedTask == processTask && !cancellationToken.IsCancellationRequested)
+                            {
+                                // Only show this error message if the process exited non-zero due to a normal process exit.
+                                // Don't show this if dotnet-watch killed the inner process due to file change or CTRL+C by the user
+                                _reporter.Error($"Application failed to start: {processTask.Exception.InnerException.Message}");
```

Comment:
Is there any risk of the `InnerException` being `null` here?

Reply:
Nope, this is always an AggregateException, so it always contains an InnerException.